bundles/org.openhab.ui/web/src/components/widgets/widget-foreground-service.js

```diff
@@ -0,0 +1,47 @@
+export default {
+  data () {
+    return {
+      pageEl: null,
+      inForeground: false
+    }
+  },
+  mounted () {
+    // determine the current page - support: https://caniuse.com/?search=closest
+    this.pageEl = this.$el.closest('.page')
+    const isPopup = this.$el.closest('.popup') !== null
```

Comment:
Could you elaborate on this? I tested the original code with popups and found out that it seemed to work as expected?

Reply:
Yeah, it was working  for me as well when i had the logic directly in my widget , but when using as a mixin, it was not finding the `page-current` class.

I actually can't test this right now to show some examples, the build seems to be broken?  I rebased to `main` and now I am getting the error when running `npm start`:

```
Module not found: Error: Can't resolve 'source-map-loader' in '/Users/daniel/openhab-main/git/openhab-webui/bundles/org.openhab.ui/web'
``` 
Along with a whole lot of other source-map-loader errors.  I tried blowing away my node modules and reinstalling, and when that did not work, i cloned a clean copy of the repo, did a `npm install` then `npm start` and am seeing the same error.

Reply:
So it looks like this line https://github.com/openhab/openhab-webui/blob/f1851e9be2c995a35b59fe6171b4b12c8859515c/bundles/org.openhab.ui/web/build/webpack.config.js#L92 which was introduced in commit https://github.com/openhab/openhab-webui/commit/f1851e9be2c995a35b59fe6171b4b12c8859515c  is causing my local build to fail, simply removing this test works FYI.

Reply:
so back to the topic at hand, if i dump out the `this.$el.closest('.page')` to the console during a popup, you can see it does not contain the `page-current` class.  I'm not sure why thats different now, maybe a loading order issue?

<img width="345" alt="image" src="https://user-images.githubusercontent.com/1903737/172064962-f7a97ac8-2fd2-42f9-b1fe-31635fc30268.png">

Reply:
>  and npm i source-map-loader should also solve the problem for the time being 

Yeah, that was the first thing i tried, but it led to some package conflict which i thought could happen when your npm cache is out of sync (it was not the issue in this case), so when trying from a fresh clone failed,  I kinda gave up debugging the issue :-)

>That would be another overlook - I'm assuming you want the widget to work when it's in a page presented as a popup.

i have only tried this PR running the widget in popups and in full pages,  but not a page in a popup (or other modals)

>So this "page-current class means the page is already is the foreground" logic is wrong

So does that mean this technique is not viable, or needs to be tweaked in your opinion?